File: nomad-samples/admin/py/basic-live-channel-1.1/live_input/get_live_inputs.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def get_live_inputs(AUTH_TOKEN):
    # Check for valid parameters
    if (not AUTH_TOKEN):
        raise Exception("Get Live Inputs: Invalid API call")


    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        # Send the request
        RESPONSE = requests.get(f"{ADMIN_URL}/liveInput", headers= HEADERS)
    
        # Return JSON response
        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, "Get Live Inputs failed")

File: nomad-samples/admin/js/basic-live-channel/live-input/get-live-inputs.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Get Live Inputs
 *
 * Note: Live Inputs are direct calls,
 *       they are not indexed, thus, the Search API cannot be used.
 *
 * @param {string} authToken    | Authorization token
 * @returns JSON Object containing a list of live inputs
 */
export default async function getLiveInputs(authToken) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/liveInput`, {
        method: "GET",
        headers: HEADERS
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        // Return JSON response
        return await RESPONSE.json();
    }

    await apiExceptionHandler(RESPONSE, "Get Live Inputs failed");
}
